service/history/historyEngine.go

```diff
@@ -723,8 +723,8 @@ func (e *historyEngineImpl) updateEntityNotExistsErrorOnPassiveCluster(err error
 		if namespaceCacheErr != nil {
 			return err // if could not access namespace cache simply return original error
 		}
-		namespaceNotActiveErr := namespaceCache.GetNamespaceNotActiveErr().(*serviceerror.NamespaceNotActive)
```

Comment:
`GetNamespaceNotActiveErr()` returns `nil` if namespace is active and it is illegal to use type assertion on `nil`.